icecube-cosmic-rays.md

```diff
@@ -0,0 +1,60 @@
+### Who am I?
+
+I'm [James Bourbeau](https://github.com/jrbourbeau). I'm a graduate student in
+the Physics department at the University of Wisconsin&mdash;Madison and a member
+of the [IceCube South Pole Neutrino Observatory Collaboration](https://icecube.wisc.edu/).
+
+
+### What problem am I trying to solve?
+
+Cosmic rays are energetic particles that originate from outer space. While they
+have been studied since the early 1900s, the sources of high-energy cosmic rays
+are still not well known. I use data collected by IceCube to study how the
+cosmic-ray flux changes with energy and particle mass. This can help provide
+valuable insight into our understanding of the origin of cosmic rays.
+
+
+### How does Dask help?
+
+My analysis pipeline consists of several steps: data preprocessing,
+training/validating machine learning models, performing statistical analyses,
+implementing custom algorithms, etc. Dask provides a user-friendly framework for
+both parallelizing each of these steps as well as seamlessly scaling to large
+amounts of data when required.
```

Comment:
If possible I'd like to extract more details from you here :)  Can I ask you to expand a bit on the parts of Dask that you use?  

For example, do you use Dask dataframe to load data from CSV files and then do data preprocessing?

Do you use dask.delayed to build custom algorithms?  If so, what kinds of algorithms?  Specificity here is good.

When you scale to large amounts of data what kind of data and how much?

Reply:
Yeah, I can definitely add some more details here 

> Do you use dask.delayed to build custom algorithms? If so, what kinds of algorithms? 

Mostly, I used dask.delayed when dealing with embarrassingly parallel problems. I can add a particular example of this. 

> When you scale to large amounts of data what kind of data and how much?

That actually varies quite a bit. Sometimes I'm only analyzing a few GB (e.g. a 3GB dataframe) and other times I'll need to work with a 100GB dataset.